geonode/api/resourcebase_api.py

```diff
@@ -273,11 +280,25 @@ def filter_bbox(self, queryset, bbox):
         returns the modified query
         """
         bbox = bbox.split(',')  # TODO: Why is this different when done through haystack?
-        bbox = list(map(str, bbox))  # 2.6 compat - float to decimal conversion
-        intersects = ~(Q(bbox_x0__gt=bbox[2]) | Q(bbox_x1__lt=bbox[0]) |
-                       Q(bbox_y0__gt=bbox[3]) | Q(bbox_y1__lt=bbox[1]))
-
-        return queryset.filter(intersects)
+        bbox = list(map(float, bbox))
+        x_min = self.x_value_wrapper(bbox[0])
+        x_max = self.x_value_wrapper(bbox[2])
+        
+        if abs(bbox[0] - bbox[2]) >= 360:
+            return Layer.objects.all()
+
+        # bbox_tuple in strict format of (xmin, ymin, xmax, ymax)
+        # as required by Polygon.from_bbox() function
+        if x_min > x_max:
+            left_bbox_tuple = (DEFAULT_WEST, bbox[1], x_max, bbox[3])
+            left_polygon = Polygon.from_bbox(left_bbox_tuple)
```

Comment:
I think you can do this in one line, `left_polygon = Polygon.from_bbox((DEFAULT_WEST, bbox[1],...))`